headers.go

```diff
@@ -416,9 +417,24 @@ func validateHeaderParameters(h map[any]any, protected bool) error {
 				return fmt.Errorf("header parameter: crit: %w", err)
 			}
 		case HeaderLabelContentType:
-			if !canTstr(value) && !canUint(value) {
+			is_tstr := canTstr(value)
+			if !is_tstr && !canUint(value) {
 				return errors.New("header parameter: content type: require tstr / uint type")
 			}
+			if is_tstr {
+				v := value.(string)
+				if len(v) == 0 {
+					return errors.New("header parameter: content type: require non-empty string")
+				}
+				if v[0] == ' ' || v[len(v)-1] == ' ' {
+					return errors.New("header parameter: content type: require no leading/trailing whitespace")
+				}
+				// Basic check that the content type is of form type/subtype.
+				// We don't check the precise definition though (RFC 6838 Section 4.2).
```

Comment:
I am OK with a quick and dirty sanity check, but it seems it wouldn't be prohibitively difficult to do the full sanitisation.  6838 `restricted-name`'s ABNF is `[a-zA-Z0-9][a-zA-Z0-9!#$&\-^_.+]{0,126}` in regex terms, so if we check that `v` matches
```
^[a-zA-Z0-9][a-zA-Z0-9!#$&\-^_.+]{0,126}/[a-zA-Z0-9][a-zA-Z0-9!#$&\-^_.+]{0,126}$
```
we'd be done?

Reply:
> we'd be done?

Nop. It can also contain parameters and subparameters. As per [RFC 6838 Section 4.3](https://www.rfc-editor.org/rfc/rfc6838#section-4.3), parameter names must conform to `restricted-name` ABNF, but the parameter value format is not specified, leaving it to each registrant.

Reply:
ah, this bit of the spec is confusing:

_"Text values follow the syntax of "\<type-name\>/\<subtype-name\>", where \<type-name\> and \<subtype-name\> are defined in [Section 4.2](https://www.rfc-editor.org/rfc/rfc6838#section-4.2) of [[RFC6838](https://www.rfc-editor.org/rfc/rfc9052#RFC6838)]._

as it seems to suggest cty would use a subset of the media type string...

But in any case, that's possibly even better from our perspective because we can use [`mime.ParseMediaType`](https://pkg.go.dev/mime#ParseMediaType)?

Reply:
> ah, this bit of the spec is confusing:

It is confusing, indeed.

> But in any case, that's possibly even better from our perspective because we can use [mime.ParseMediaType](https://pkg.go.dev/mime#ParseMediaType)?

We could be being too restrictive by using that method, as MIME is not the only allowed content type format, and it might impose additional restrictions, as noted in RFC 6838 Section 4.2:

> Additionally, various protocols, including but not limited to HTTP and MIME, MAY impose additional restrictions on the media types they can transport.  (See [[RFC2046](https://www.rfc-editor.org/rfc/rfc2046)] for additional information on the restrictions MIME imposes.)

Reply:
> > ah, this bit of the spec is confusing:
> 
> It is confusing, indeed.

😄 

> > But in any case, that's possibly even better from our perspective because we can use [mime.ParseMediaType](https://pkg.go.dev/mime#ParseMediaType)?
> 
> We could be being too restrictive by using that method, as MIME is not the only allowed content type format,

note that `mime.ParseMediaType` also does HTTP (or so it says)

> and it might impose additional restrictions, as noted in RFC 6838 Section 4.2:
> 
> > Additionally, various protocols, including but not limited to HTTP and MIME, MAY impose additional restrictions on the media types they can transport.  (See [[RFC2046](https://www.rfc-editor.org/rfc/rfc2046)] for additional information on the restrictions MIME imposes.)

While theoretically true, MIME and HTTP really cover *a lot* of ground -- i.e., I'd be surprised if we hit a case that'd need special treatment.

Anyway, I should have made clearer that this is non-blocking 👍

Reply:
Thanks, @thomas-fossati for the dialog and non-blocking comment.